src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -84,7 +84,9 @@ public String getIconFilePathPattern() {
     /**
      * Creates a default navigator, if there should be one displayed.
      * May be interpreted by {@code jenkins.branch.SpecificDescriptor}.
-     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}.
+     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}
+     * and implement {@link SCMNavigatorDescriptor#getDescription()}.
```

Comment:
…though IIRC in Jenkins 2, `newInstanceDetail` is only loaded for compatibility in case `getDescription` is not overridden, so we could perhaps ask the implementor to override `getDescription` and not even mention `newInstanceDetail`.